### PR TITLE
Fixes #11034 - custom uniqueness validations for interface attributes

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by domcleal** [comment]
Please note that the PR introduces many test failures.

**Comment by ares** [comment]
[test], Katello failures seems unrelated

File: app/models/nic/base.rb [hunk 1]
```diff
@@ -219,8 +219,19 @@ def validate_host_organization
...
     end
 
+    def mac_uniqueness
+      interface_attribute_uniqueness(:mac)
+    end
+
     private
 
+    def interface_attribute_uniqueness(attr, base = Nic::Base.scoped)
+      in_memory_candidates = self.host.present? ? self.host.interfaces.select { |i| i.persisted? && !i.marked_for_destruction? } : [self]
```
Comment (dLobatog) [inline app/models/nic/base.rb:229]: Very neat and readable :smile:

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -920,7 +920,7 @@ def build_status_label(options = {})
   # because the validation happens before transaction is committed, so data are not in DB
   # yet, this is the reason why we "reimplement" uniqueness validation
   def validate_dns_name_uniqueness
-    dups = self.interfaces.group_by { |i| [ i.name, i.domain_id ] }.detect { |dns, nics| dns.first.present? && nics.count > 1 }
+    dups = self.interfaces.select { |i| !i.marked_for_destruction? }.group_by { |i| [ i.name, i.domain_id ] }.detect { |dns, nics| dns.first.present? && nics.count > 1 }
```
Comment (dLobatog) [inline app/models/host/managed.rb:923]: Consider extracting this to a separate method, I'll merge for the moment though

**Comment by dLobatog** [comment]
Merged as 12f3859d99d5db778e455ca31a3407346c20d020, thanks @ares!

**Pull request closed without merging**